### ISSUE TITLE
Core dumps possibly related to AVX after upgrading to 2.3.14

### ISSUE THREAD

**Issue by Jehops** [veg/hyphy#835]
We are seeing core dumps on our older nodes.  Is the cmake argument, NOAVX, doing what it should do?  We are still seeing `-march=corei7-avx -mtune=corei7-avx` with `-DNOAVX=ON`.

```
cmake -DNOAVX=ON .
-- The C compiler identification is Clang 6.0.0
-- The CXX compiler identification is Clang 6.0.0
-- Check for working C compiler: /usr/bin/cc
-- Check for working C compiler: /usr/bin/cc -- works
-- Detecting C compiler ABI info
-- Detecting C compiler ABI info - done
-- Detecting C compile features
-- Detecting C compile features - done
-- Check for working CXX compiler: /usr/bin/c++
-- Check for working CXX compiler: /usr/bin/c++ -- works
-- Detecting CXX compiler ABI info
-- Detecting CXX compiler ABI info - done
-- Detecting CXX compile features
-- Detecting CXX compile features - done
-- Performing Test HAVE_AVX_EXTENSIONS
-- Performing Test HAVE_AVX_EXTENSIONS - Success
Set compiler flags to -fsigned-char -O3 -march=corei7-avx -mtune=corei7-avx
-- Looking for amd_log in libamdlibm.so
-- Looking for amd_log in libamdlibm.so - not found
-- Could NOT find OpenMP_C (missing: OpenMP_C_FLAGS OpenMP_C_LIB_NAMES)
-- Could NOT find OpenMP_CXX (missing: OpenMP_CXX_FLAGS OpenMP_CXX_LIB_NAMES)
-- Could NOT find OpenMP (missing: OpenMP_C_FOUND OpenMP_CXX_FOUND)
-- Found CURL: /usr/local/lib/libcurl.so (found version "7.61.0")
-- Found PythonInterp: /usr/local/bin/python2.7 (found version "2.7.15")
-- Looking for pthread.h
-- Looking for pthread.h - found
-- Looking for pthread_create
-- Looking for pthread_create - not found
-- Looking for pthread_create in pthreads
-- Looking for pthread_create in pthreads - not found
-- Looking for pthread_create in pthread
-- Looking for pthread_create in pthread - found
-- Found Threads: TRUE
-- Could NOT find OpenCL (missing: OPENCL_LIBRARIES OPENCL_INCLUDE_DIR)
-- Could NOT find MPI_C (missing: MPI_C_LIB_NAMES MPI_C_HEADER_DIR MPI_C_WORKS)
-- Could NOT find MPI_CXX (missing: MPI_CXX_LIB_NAMES MPI_CXX_HEADER_DIR MPI_CXX_WORKS)
-- Could NOT find MPI (missing: MPI_C_FOUND MPI_CXX_FOUND)
-- Configuring done
-- Generating done
```

**Comment by spond** [comment]
Dear @jehops,

I can never remember how to activate cmake flags, but let me check to make sure that the configuration file is not buggy. 

Best,
Sergei

**Comment by spond** [comment]
Dear @Jehops,

Please pull the `beta` branch where this issue (not checking for NOAVX for Clang) has been fixed.

Best,
Sergei

**Comment by Jehops** [comment]
Looks good.  The [build log](http://pkg.awarnach.mathstat.dal.ca/data/11amd64-default/2018-08-13_07h17m27s/logs/hyphy-2.3.14.log) now says `Set compiler flags to -fsigned-char -O3 -mno-avx -msse3 ` and  `HYPHYMP` no longer crashes on our older nodes.

I will skip updating the FreeBSD package to  2.3.14 and wait until these fixes hit a new release.

Thank you,
Joseph

**Comment by spond** [comment]
Dear @Jehops,

Sounds good; we'll prioritize 2.3.15 with this fix (and others), because 2.3.14 has some substantial bug fixes and analysis improvements, so it is a recommended update.

Best,
Sergei

@stevenweaver
